File: src/main/java/com/github/liebharc/queryenrichment/ParameterlessEnrichment.java
package com.github.liebharc.queryenrichment;


/**
 * An {@link Enrichment} with no paramters.
 * @param <TAttribute> Attribute type.
 */
public abstract class ParameterlessEnrichment<TAttribute> extends Enrichment<TAttribute, Object> {
    private static final long serialVersionUID = -3632157624255773985L;

    public ParameterlessEnrichment(Attribute<TAttribute> attribute, Dependency dependency) {
        super(attribute, dependency);
    }

    public ParameterlessEnrichment(Attribute<TAttribute> attribute, String columnOrNull, Dependency dependency) {
        super(attribute, columnOrNull, dependency);
    }

    @Override
    public final void enrich(IntermediateResult result, Object parameter) {
        this.enrich(result);
    }

    /**
     * As {@link #enrich(IntermediateResult, Object)} without passing a parameter.
     */
    public abstract void enrich(IntermediateResult result);
}

File: src/main/java/com/github/liebharc/queryenrichment/ParameterlessEnrichment.kt
package com.github.liebharc.queryenrichment


/**
 * An [Enrichment] with no paramters.
 * @param <TAttribute> Attribute type.
</TAttribute> */
abstract class ParameterlessEnrichment<TAttribute>(attribute: Attribute<TAttribute>, dependency: Dependency) : Enrichment<TAttribute, Any?>(attribute, dependency) {

    override fun enrich(result: IntermediateResult, parameter: Any?) {
        this.enrich(result)
    }

    /**
     * As [.enrich] without passing a parameter.
     */
    abstract fun enrich(result: IntermediateResult)

    companion object {
        private const val serialVersionUID = -3632157624255773985L
    }
}
